lib/permissions.js

```diff
@@ -529,16 +523,57 @@ exports = module.exports = {
             }
           }
 
-          roles.forEach(function (r) {
-            r.statement.forEach(function (s, index) {
-              s.resource = s.resource.replace('{{tenantid}}', tenantId);
-              statements.push(s);
-            });
-          });
+          if (!roles || 0 === roles.length) {
+            return callback(null, []);
+          }
 
-          return callback(null, statements);
-        });
-      });
+          async.each(
+            roles,
+            function (r, nextRole) {
+              async.each(
+                r.statement,
+                function (s, nextStatement) {
+                  s.resource = s.resource.replace('{{tenantid}}', tenantId);
+                  statements.push(s);
+                  nextStatement();
+                },
+                nextRole
```

Comment:
Ahhh yeah, I didn't realise it was being passed to the nested async.each, thought you meant to call it there and then.